src/ShowingTransitionSupport.js

```diff
@@ -0,0 +1,221 @@
+	* @default undefined
+	* @public
+	*/
+	hiddenMethod: undefined,
+
+	/**
+	* The the classname to apply for the "shown" (component is visible) resting state.
+	*
+	* @type {String}
+	* @default undefined
+	* @public
+	*/
+	shownClass: undefined,
+
+	/**
+	* The the classname to apply for the "hidden" (component is not visible) resting state.
+	*
+	* @type {String}
+	* @default 'hidden'
+	* @public
+	*/
+	hiddenClass: undefined,
+
+	/**
+	* The the classname to apply for the "hiding" (component has started the transition to the
+	* hidden state) transition state.
+	*
+	* @type {String}
+	* @default 'hiding'
+	* @public
+	*/
+	hidingClass: undefined,
+
+	/**
+	* The the classname to apply for the "showing" (component has started the transition to the
+	* shown state) transition state.
+	*
+	* @type {String}
+	* @default 'showing'
+	* @public
+	*/
+	showingClass: undefined,
+
+	/**
+	* Initializes the defaults, and prepares the component with its classes in case of initially
+	* "showing: false".
+	*
+	* @method
+	* @private
+	*/
+	create: kind.inherit(function (sup) {
+		return function () {
+			sup.apply(this, arguments);
+			this.showingDuration = (this.showingDuration === undefined) ? null      : this.showingDuration;
+			this.hidingDuration  = (this.hidingDuration  === undefined) ? null      : this.hidingDuration;
+			this.shownMethod     = (this.shownMethod     === undefined) ? null      : this.shownMethod;
+			this.hiddenMethod    = (this.hiddenMethod    === undefined) ? null      : this.hiddenMethod;
+			this.shownClass      = (this.shownClass      === undefined) ? 'shown'   : this.shownClass;
+			this.hiddenClass     = (this.hiddenClass     === undefined) ? 'hidden'  : this.hiddenClass;
+			this.hidingClass     = (this.hidingClass     === undefined) ? 'hiding'  : this.hidingClass;
+			this.showingClass    = (this.showingClass    === undefined) ? 'showing' : this.showingClass;
+			this.showingChanged();
+		};
+	}),
+
+	/**
+	* Overrides the showingChanged handler to add support for transitions at the right times and
+	* places.
+	*
+	* @method
+	* @private
+	*/
+	showingChanged: kind.inherit(function (sup) {
+		return function (sender, ev) {
+			var args = arguments;
+
+			// Prepare our visual state
+			this.applyStyle('display', null);
+			this.applyStyle('visibility', null);
+			if (this.showing) {
+				// Reset our state classes, in case we switched mid-stream
+				this.removeClass(this.hidingClass);
+				this.removeClass(this.hiddenClass);
+				sup.apply(this, args);
+				if (this.showingDuration && this.generated) {
+					this.set('showingTransitioning', true);
+					// Start transition: Apply a class and start a timer.
+					// When timer finishes, run the exit function,
+					// remove the transitioning class
+					// and add the final-state class
+					this.addClass(this.showingClass);
+					this.startJob('showingTransition', function () {
+						utils.call(this, this.shownMethod);	// Run the supplied method.
```

Comment:
So something like a new property that allows you to specify the scope, or something fancier? I assume you're meaning in the 2nd to last word "the `this` control" meaning the possessor of the components block; and not literally english "this", because english "this" would refer to the control that the mixin is mixed into, and we should not be putting functions on the components in the components block.

Anyway, what did you have in mind for making the `utils.call` method more versatile, instead of directly scoping it to `this`?